packages/package.go

```diff
@@ -216,9 +216,44 @@ func getDownloadPath(p Package, t string) string {
 	return path.Join("/epr", p.Name, p.Name+"-"+p.Version+".zip")
 }
 
+// MustParsePackage creates a new package instances based on the given base path.
+// The path passed goes to the root of the package where the manifest.yml is.
+// It runs more strict validation than NewPackage, e.g. ensuring that all categories are valid.
+func MustParsePackage(logger *zap.Logger, basePath string, fsBuilder FileSystemBuilder) (*Package, error) {
```

Comment:
Does this method still require a logger? Where we intend to use it there won't be one, right?

Reply:
I added this logger here because it was required for this: 
https://github.com/elastic/package-registry/blob/54c21f2ab37d4a6e601d4a718f9b5a0cde3139a0/packages/package.go#L246

With the changes proposed here, this parameter could be removed and just used the logger in `NewPackage` function without requiring to be added as a field in `Package`.

Reply:
Applied change to remove logger in https://github.com/elastic/package-registry/pull/1333/commits/7dc1e70b3790efe4d07575fbe07675d5433c2418